conjure-java-core/src/test/resources/example-unions.yml

```diff
@@ -0,0 +1,25 @@
+types:
+  definitions:
+    default-package: com.palantir.product
+    objects:
+      SimpleUnion:
+        union:
+          foo: string
```

Comment:
SInce we're no longer suffixing the class with `Wrapper`, what happens if we have a field with the same name as the union name (`simpleUnion` in this case)? We probably need to sanitize that.